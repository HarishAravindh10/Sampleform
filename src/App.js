import React from 'react';
import { useState } from 'react';


function App() {
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const [errorMessage, setErrorMessage] = useState('');

  const handleSubmit = (event) => {
    event.preventDefault();

    // Simulate login validation (replace with actual authentication logic)
    if (username !== 'admin' || password !== 'password') {
      setErrorMessage('Invalid username or password');
      return;
    }

    // Handle successful login (e.g., redirect to another page)
    console.log('Login successful!');
  };

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="username">Username:</label>
      <input
        type="text"
        id="username"
        value={username}
        onChange={(e) => setUsername(e.target.value)}
      />
      <label htmlFor="password">Password:</label>
      <input
        type="password"
        id="password"
        value={password}
        onChange={(e) => setPassword(e.target.value)}
      />
      <button type="submit">Login</button>
      {errorMessage && <p className="error">{errorMessage}</p>}
    </form>
  );
}

export default App;