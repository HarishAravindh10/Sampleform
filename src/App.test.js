import React from 'react';
import App from './App';
import renderer from 'react-test-renderer';

describe('App', () => {
  it('should render username and password input fields', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it('should update password state on input change', () => {
  //   const component = renderer.create(<App />);
  //   const instance = component.getInstance();

  //   instance.setState({ password: 'password' });

  //   const tree = component.toJSON();
  //   expect(tree).toMatchSnapshot();
  // });

  // Add more tests for other functionalities (optional)
});