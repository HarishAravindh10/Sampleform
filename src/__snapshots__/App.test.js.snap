// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render username and password input fields 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="username"
  >
    Username:
  </label>
  <input
    id="username"
    onChange={[Function]}
    type="text"
    value=""
  />
  <label
    htmlFor="password"
  >
    Password:
  </label>
  <input
    id="password"
    onChange={[Function]}
    type="password"
    value=""
  />
  <button
    type="submit"
  >
    Login
  </button>
</form>
`;

exports[`renders learn react link 1`] = `
<div
  className="App"
>
  Learn React
</div>
`;
